tools/clang/include/clang/SPIRV/SpirvInstruction.h

```diff
@@ -1943,6 +1978,730 @@ class SpirvDemoteToHelperInvocationEXT : public SpirvInstruction {
   bool invokeVisitor(Visitor *v) override;
 };
 
+/// \breif Base class for all OpenCL.DebugInfo.100 extension instructions.
+/// Note that all of these instructions should be added to the SPIR-V module as
+/// an OpExtInst instructions. So, all of these instructions must:
+/// 1) contain the result-id of the extended instruction set
+/// 2) have OpTypeVoid as their Result Type.
+/// 3) contain additional QualType and SpirvType for the debug type.
+class SpirvDebugInstruction : public SpirvInstruction {
+public:
+  static bool classof(const SpirvInstruction *inst) {
+    return inst->getKind() >= IK_DebugInfoNone &&
+           inst->getKind() <= IK_DebugTypeTemplateParameter;
+  }
+
+  void setDebugType(SpirvDebugInstruction *type) { debugType = type; }
+  void setInstructionSet(SpirvExtInstImport *set) { instructionSet = set; }
+  SpirvExtInstImport *getInstructionSet() const { return instructionSet; }
+  uint32_t getDebugOpcode() const { return debugOpcode; }
+  QualType getDebugQualType() const { return debugQualType; }
+  const SpirvType *getDebugSpirvType() const { return debugSpirvType; }
+  SpirvDebugInstruction *getDebugType() const { return debugType; }
+  void setDebugQualType(QualType type) { debugQualType = type; }
+  void setDebugSpirvType(const SpirvType *type) { debugSpirvType = type; }
+
+  virtual SpirvDebugInstruction *getParentScope() const { return nullptr; }
+
+protected:
+  // TODO: Replace opcode type with an enum, when it is available in
```

Comment:
Please file an issue for replacing all usages of uint32_t for flags with enums from SPIRV-Headers.